### PR TITLE
feat(payload) : optimize new payload job by fetching only header hash instead of block

### PR DESCRIPTION
**Pull request by lean-apple**
Optimization in response of this [comment.](https://github.com/paradigmxyz/reth/issues/12103#issuecomment-2444740389)

### CODE REVIEW

File: crates/payload/basic/src/lib.rs [hunk 1]
```diff
@@ -144,33 +144,30 @@ where
...
-                .ok_or_else(|| PayloadBuilderError::MissingParentBlock(attributes.parent()))?;
-
-            // we already know the hash, so we can seal it
-            block.seal(attributes.parent())
+            // Fetch specific header by hash
+            self.client
+                .sealed_header_by_hash(attributes.parent())
+                .map_err(PayloadBuilderError::from)?
+                .ok_or_else(|| PayloadBuilderError::MissingParentHeader(attributes.parent()))?
         };
```
Comment (lean-apple) [inline crates/payload/basic/src/lib.rs:159]: @hai-rise

**Review by mattsse — APPROVED** [review]
cool, this is great

File: crates/payload/basic/src/lib.rs [hunk 2]
```diff
@@ -144,33 +144,30 @@ where
...
-                .client
-                .find_block_by_hash(attributes.parent(), BlockSource::Any)?
-                .ok_or_else(|| PayloadBuilderError::MissingParentBlock(attributes.parent()))?;
-
-            // we already know the hash, so we can seal it
-            block.seal(attributes.parent())
+            // Fetch specific header by hash
+            self.client
+                .sealed_header_by_hash(attributes.parent())
+                .map_err(PayloadBuilderError::from)?
```
Comment (hai-rise) [inline crates/payload/basic/src/lib.rs:157]: We don't seem to need these `map_err`s?

File: crates/payload/basic/src/lib.rs [hunk 3]
```diff
@@ -144,33 +144,30 @@ where
...
         };
 
-        let hash = parent_block.hash();
-        let parent_header = parent_block.header();
-        let header = SealedHeader::new(parent_header.clone(), hash);
-
-        let config =
-            PayloadConfig::new(Arc::new(header), self.config.extradata.clone(), attributes);
+        let config = PayloadConfig::new(
+            Arc::new(parent_header.clone()),
```
Comment (hai-rise) [inline crates/payload/basic/src/lib.rs:162]: Nit but we won't need this `clone` if we use an intermediate `hash` like previously, or simply create `cached_reads` before `config` 🙏.

File: crates/payload/basic/src/lib.rs (hunk 1 above)
Comment (hai-rise) [reply crates/payload/basic/src/lib.rs:159]: Thank you!!

**Pull request merged**